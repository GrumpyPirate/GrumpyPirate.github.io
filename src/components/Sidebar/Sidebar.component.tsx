import React, { FunctionComponent } from 'react';
import { Link } from 'react-router-dom';
import classnames from 'classnames';

import Sitenav from 'components/Sitenav/Sitenav.container';
import Container from 'components/Layout/Container/Container.component';
import ScreenreaderOnlyText from 'components/Accessibility/ScreenreaderOnlyText/ScreenreaderOnlyText.component';

import { SidebarProps } from './Sidebar.component.types';
import classes from './Sidebar.component.scss';

const Sidebar: FunctionComponent<SidebarProps> = ({
  isMobileNavigationOpen,
  toggleMobileNavigation,
}: SidebarProps) => (
  <aside className={classes['sidebar']}>
    <Container>
      <div className={classes['sidebar__content']}>
        <Link to="/" className={classes['sidebar__brand']}>
          <img className={classes['sidebar__brand__image']} alt="" width="52" height="52" />
          <span className={classes['sidebar__brand__name']}>Edward Cobbold</span>
        </Link>
        <span className={classes['sidebar__brand__job-title']}>Frontend Developer</span>

        <hr className={classes['sidebar__divider']} />

        <div className={classes['sidebar__btn-sitenav-container']}>
          <button
            type="button"
            className={classes['sidebar__btn-sitenav']}
            onClick={toggleMobileNavigation}
            aria-expanded={isMobileNavigationOpen}
          >
            <span className={classes['burger-bar']} />
            <ScreenreaderOnlyText text="Toggle menu" />
          </button>
        </div>

        <div
          className={classnames(
            classes['sidebar__sitenav-wrapper'],
            classes['sidebar__sitenav-wrapper--desktop'],
          )}
        >
          <Sitenav />
        </div>
      </div>
    </Container>

    <menu
      className={classnames(
        classes['sidebar__sitenav-wrapper'],
        classes['sidebar__sitenav-wrapper--mobile'],
        {
          [classes['sidebar__sitenav-wrapper--mobile--is--closed']]: !isMobileNavigationOpen,
        },
      )}
    >
      <Container>
        <Sitenav />
      </Container>
    </menu>
  </aside>
);

export default Sidebar;
