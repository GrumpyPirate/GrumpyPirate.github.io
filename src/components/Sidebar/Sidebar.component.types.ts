export interface SidebarProps {
  isMobileNavigationOpen: boolean;
  toggleMobileNavigation: () => void;
}
