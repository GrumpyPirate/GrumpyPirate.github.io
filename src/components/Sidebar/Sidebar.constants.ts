import Color from 'color';
import styled, { css } from 'styled-components';

import spriteImage from 'images/head-sprite.png';
import {
  animDur,
  buttonReset,
  createHeading,
  fontSizeH4,
  gridGutterWidthBase,
  mediaQueries,
  noOutline,
  palette,
  rem,
  sidebarHeightMobile,
} from 'styles';

const burgerBarSize = 32;
const burgerBarHeight = 1;
const burgerBarSpacing = 8;

export const Content = styled.div`
  align-items: center;
  display: flex;
  height: ${rem(sidebarHeightMobile)};
  justify-content: space-between;
  padding-bottom: ${rem(gridGutterWidthBase / 4)};
  padding-top: ${rem(gridGutterWidthBase / 4)};
  position: relative;
  z-index: 11;

  @media ${mediaQueries.lg} {
    background: 0;
    display: block;
    height: auto;
    padding-bottom: 0;
    padding-top: 0;
  }
`;

export const BrandLogo = styled.figure`
  ${noOutline}
  background: url(${spriteImage}) top left no-repeat transparent;
  background-size: ${rem(32)} ${rem(64)};
  display: block;
  filter: drop-shadow(${rem(4)} ${rem(4)} ${rem(2)} rgba(0, 0, 0, 0.2));
  height: ${rem(32)};
  image-rendering: crisp-edges;
  margin: 0;
  transition: transform ${animDur}s;
  width: ${rem(32)};

  @media ${mediaQueries.lg} {
    background-size: ${rem(48)} ${rem(96)};
    height: ${rem(48)};
    margin-left: auto;
    width: ${rem(48)};
  }
`;

export const Brand = styled.div`
  a {
    ${noOutline};
    ${createHeading(5)}

    display: flex;
    align-items: center;
    letter-spacing: normal;
    margin-bottom: 0;

    @media ${mediaQueries.lg} {
      font-size: ${rem(fontSizeH4)};
      display: block;
    }

    &,
    &:hover,
    &:focus {
      color: ${palette.themeLightShades};
    }

    &:hover {
      ${BrandLogo} {
        background-position: bottom left;
      }
    }

    &:active {
      ${BrandLogo} {
        transform: scale(0.875);
        transition-duration: 0s;
      }
    }
  }
`;

export const BrandName = styled.span`
  color: inherit;
  display: inline-block;
  margin-left: 1rem;
  vertical-align: middle;
`;

export const JobTitle = styled.span`
  color: ${palette.themeLightShades};
  display: block;
  font-size: 0.75em;
  opacity: 0.75;

  @media ${mediaQueries.mdDown} {
    display: none;
  }
`;

export const Divider = styled.hr`
  border-color: ${Color(palette.themeLightShades).alpha(0.2).string()};
  border-style: solid none none;
  border-width: 1px;
  margin: 2rem 0;

  @media ${mediaQueries.mdDown} {
    display: none;
  }
`;

export const BurgerButtonWrapper = styled.div`
  @media ${mediaQueries.md} {
    display: none;
  }
`;

export const BurgerButtonBar = styled.span`
  top: calc(50% - ${rem(burgerBarHeight / 2)});

  &,
  &::before,
  &::after {
    background-color: ${palette.themeLightShades};
    display: block;
    height: ${rem(burgerBarHeight)};
    left: 0;
    position: absolute;
    transition: transform ${animDur}s ease, background-color ${animDur}s ease;
    width: ${rem(burgerBarSize)};
  }

  &::before,
  &::after {
    content: '';
  }

  &::before {
    transform: translateY(-${rem(burgerBarHeight + burgerBarSpacing)});
  }

  &::after {
    transform: translateY(${rem(burgerBarHeight + burgerBarSpacing)});
  }
`;

export const BurgerButton = styled.button`
  ${buttonReset}

  display: block;
  height: ${rem(burgerBarSize)};
  position: relative;
  transition: transform 0.075s ease;
  width: ${rem(burgerBarSize)};

  &:active {
    transform: scale(0.9);
  }

  &[aria-expanded='true'] {
    ${BurgerButtonBar} {
      background-color: transparent;

      &::before {
        transform: rotate(-135deg);
      }

      &::after {
        transform: rotate(135deg);
      }
    }
  }
`;

export const DesktopSitenavWrapper = styled.div`
  @media ${mediaQueries.smDown} {
    display: none;
  }
`;

export const MobileSitenavWrapper = styled.div<{ isOpen: boolean }>`
  background-color: ${palette.themeMainBrand};
  border-bottom: solid ${rem(1)} ${palette.themeAccentLight};
  border-top: solid ${rem(1)} ${Color(palette.themeAccentLight).alpha(0.25).string()};
  left: 0;
  margin: 0;
  padding: 0.5rem 0;
  position: fixed;
  top: ${rem(sidebarHeightMobile)};
  transition: opacity ${animDur}s ease, transform ${animDur}s ease, visibility ${animDur}s ease;
  width: 100%;
  z-index: 10;

  @media ${mediaQueries.md} {
    display: none;
  }

  ${({ isOpen }) =>
    !isOpen &&
    css`
      opacity: 0;
      pointer-events: none;
      transform: translateY(-50%);
      visibility: hidden;
    `}
`;
