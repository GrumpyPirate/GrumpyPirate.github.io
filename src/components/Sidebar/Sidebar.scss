@import 'functions';
@import 'mixins';
@import 'variables';

.sidebar {
  @include make-link-colour(palette(theme-light), $hover-colour: palette(theme-light), $active-colour: palette(white), $self: false);
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  width: 100%;
  background-color: palette(theme-accent-1);
  // border-bottom: solid $hr-border-width $sidebar-border-color;
  text-transform: uppercase;
  color: palette(theme-light);
  z-index: $sidebar-zindex;

  // lg up
  @include media-breakpoint-up(lg) {
    position: relative;
    height: 100vh;
    overflow-y: auto;
    touch-action: pan-y;
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    flex: 0 0 $sidebar-width-lg;
    padding: 0;
    border: 0;
    // border-right: solid $hr-border-width $sidebar-border-color;
  } // /lg up

  // xl up
  @include media-breakpoint-up(xl) {
    flex-basis: $sidebar-width;
  } // /xl up

  &__content {
    position: relative;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding-top: $grid-gutter-width-base/ 4;
    padding-bottom: $grid-gutter-width-base/ 4;
    background-color: palette(theme-accent-1);
    height: $sidebar-height-mobile;
    z-index: $sidebar-zindex;

    // lg up
    @include media-breakpoint-up(lg) {
      display: block;
      padding-top: 0;
      padding-bottom: 0;
      background: 0;
      height: auto;
    } // /lg up
  } // /&__content

  // .sitenav {
  // } // /.sitenav__nav

  .sitenav__nav-item {
    // md only
    @include media-breakpoint-only(md) {
      display: inline-block;

      &:not(:last-child) {
        margin-right: 1rem;
      } // /&:not(:last-child)
    } // /md only
  } // /.sitenav__nav-item

  .sitenav__nav-link {
    @include lasers($direction: out, $display: block);
    font-family: $headings-font-family;
    font-weight: weight(light);
    display: block;
    padding: .25rem 0;
    margin: .25rem 0;

    // md up
    @include media-breakpoint-up(md) {
      padding: 0;
      margin: 0;
    } // /md up

    // lg up
    @include media-breakpoint-up(lg) {
      display: inline-block;
    } // /lg up
  } // /.sitenav__nav-link

  // Brand (Logo/Text)
  // -----------------------------------------------------------------------------------------------------------------
  &__brand {
    font-family: $headings-font-family;
    color: palette(theme-light);
    display: block;
    font-weight: weight(light);
    font-size: $font-size-h5;
    line-height: $headings-line-height;

    // lg up
    @include media-breakpoint-up(lg) {
      font-size: $font-size-h4;
    } // /lg up

    &__image {
      max-width: rem(32px);
      height: auto;
      display: inline-block;
      border-radius: 50%;
      filter: drop-shadow(#{$filter-shadow-right-down-sm});

      // lg up
      @include media-breakpoint-up(lg) {
        display: block;
        margin-left: auto;
        max-width: none;
      } // /lg up
    } // /&__image

    &__name {
      display: inline-block;
      vertical-align: middle;
      color: inherit;
    } // /&__name

    &__job-title {
      display: block;
      font-size: rem(12px);
      color: palette(theme-light);
      opacity: .65;
    } // /&__job-title
  } // /&__brand

  // Menu
  // -------------------------------------------------------------------------------------------------
  &__btn-sitenav {
    $burger-bar-width: 1.75rem;
    $burger-bar-height: 1px;
    $burger-bar-spacing: 7px;
    @include button-reset;
    @include no-outline;
    position: relative;
    display: block;
    width: $burger-bar-width;
    height: ($burger-bar-height * 3) + ($burger-bar-spacing * 2);
    transition: transform ($anim-dur / 2) ease;

    // Clicked
    &:active {
      transform: scale(#{1 - (1 / 10)});
    } // /&:active

    // Menu active
    &[aria-expanded='true'] {
      .burger-bar {
        background-color: transparent;

        &::before {
          transform: rotate(-135deg);
        } // /&::before

        &::after {
          transform: rotate(135deg);
        } // /&::before
      } // /.burger-bar
    } // /&[aria-expanded='true']

    .burger-bar {
      top: calc(50% - #{($burger-bar-height / 2)});

      &,
      &::before,
      &::after {
        display: block;
        width: $burger-bar-width;
        height: $burger-bar-height;
        position: absolute;
        left: 0;
        background-color: palette(theme-light);
        transition: transform $anim-dur ease,
                    background-color $anim-dur ease;
      } // /all bars

      &::before,
      &::after {
        content: '';
      }

      &::before {
        transform: translateY(-#{$burger-bar-height + $burger-bar-spacing});
        // transform-origin: bottom left;
      } // /&::before

      &::after {
        transform: translateY(#{$burger-bar-height + $burger-bar-spacing});
        // transform-origin: top left;
      } // /&::before
    } // /.burger-bar
  } // /&__btn-sitenav

  &__menu {
    position: fixed;
    top: $sidebar-height-mobile;
    left: 0;
    width: 100%;
    margin: 0;
    padding: .5rem 0;
    background-color: palette(theme-accent-1);
    border-bottom: solid $hr-border-width $sidebar-border-color;
    z-index: $sidebar-zindex - 1;
    transition: opacity $anim-dur ease,
                transform $anim-dur ease,
                visibility $anim-dur ease;

    &.is--closing,
    &.is--closed {
      opacity: 0;
      visibility: hidden;
      pointer-events: none;
      transform: translateY(-50%);
    } // /&.is--closing

    &.is--open {
      opacity: 1;
      visibility: visible;
      transform: translateY(0);
    } // /&.is--open
  } // /&__menu
} // /.sidebar
