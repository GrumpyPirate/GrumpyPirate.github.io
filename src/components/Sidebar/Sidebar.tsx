import React, { FunctionComponent } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

import ScreenreaderOnlyText from 'components/Accessibility/ScreenreaderOnlyText/ScreenreaderOnlyText';
import { Container } from 'components/Grid';
import Sitenav from 'components/Sitenav/Sitenav';
import { AppDispatch, RootState } from 'store';
import { toggleMobileNavigation } from 'store/ui/ui.actions';
import { mediaQueries, palette, rem, sidebarWidth, sidebarWidthXl } from 'styles';
import { ClassNameProps } from 'types/common';

import {
  Brand,
  BrandLogo,
  BrandName,
  BurgerButton,
  BurgerButtonBar,
  BurgerButtonWrapper,
  Content,
  DesktopSitenavWrapper,
  Divider,
  JobTitle,
  MobileSitenavWrapper,
} from './Sidebar.constants';

const Sidebar: FunctionComponent<ClassNameProps> = ({ className }) => {
  const isMobileNavigationOpen = useSelector((state: RootState) => state.ui.isMobileNavigationOpen);
  const dispatch: AppDispatch = useDispatch();

  return (
    <aside className={className}>
      <Container>
        <Content>
          <Brand>
            <Link to="/">
              <BrandLogo />
              <BrandName>Edward Cobbold</BrandName>
            </Link>
          </Brand>
          <JobTitle>Frontend Developer</JobTitle>

          <Divider />

          <BurgerButtonWrapper>
            <BurgerButton
              type="button"
              onClick={() => {
                dispatch(toggleMobileNavigation());
              }}
              aria-expanded={isMobileNavigationOpen}
            >
              <BurgerButtonBar />
              <ScreenreaderOnlyText text="Toggle menu" />
            </BurgerButton>
          </BurgerButtonWrapper>

          <DesktopSitenavWrapper>
            <Sitenav />
          </DesktopSitenavWrapper>
        </Content>
      </Container>

      <MobileSitenavWrapper isClosed={!isMobileNavigationOpen}>
        <Container>
          <Sitenav />
        </Container>
      </MobileSitenavWrapper>
    </aside>
  );
};

export default styled(Sidebar)`
  background-color: ${palette.themeMainBrand};
  color: ${palette.themeLightShades};
  left: 0;
  position: fixed;
  right: 0;
  text-transform: uppercase;
  top: 0;
  width: 100%;
  z-index: 11;

  @media ${mediaQueries.lg} {
    align-items: center;
    border: 0;
    display: flex;
    flex-wrap: wrap;
    height: 100vh;
    padding: 0;
    position: fixed;
    text-align: right;
    width: ${rem(sidebarWidth)};
  }

  @media ${mediaQueries.xl} {
    width: ${rem(sidebarWidthXl)};
  }
`;
