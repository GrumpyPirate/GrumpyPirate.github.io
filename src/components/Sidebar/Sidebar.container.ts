import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { RootState } from 'store';
import { toggleMobileNavigation } from 'store/ui/ui.actions';

import Sidebar from './Sidebar.component';
import { SidebarProps } from './Sidebar.component.types';

const mapStateToProps = (state: RootState): Pick<SidebarProps, 'isMobileNavigationOpen'> => ({
  isMobileNavigationOpen: state.ui.isMobileNavigationOpen,
});

const mapDispatchToProps = { toggleMobileNavigation };

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Sidebar));
