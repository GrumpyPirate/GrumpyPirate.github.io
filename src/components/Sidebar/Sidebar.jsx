// React
import React from 'react'

// Routing
import { NavLink, Link } from 'react-router-dom'

// Components
import Sitenav from 'components/Sitenav/Sitenav.jsx'

// SCSS
import './Sidebar.scss'

// Images
import Logo from 'images/avatar-ed--head-only.png'

class Sidebar extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      navMenuOpen: false
    }

    // Lexical this binding
    this.toggleMenu = this.toggleMenu.bind(this)
  } // /constructor(props)

  toggleMenu() {
    this.setState({
      navMenuOpen: !this.state.navMenuOpen
    })
  } // /openMenu()

  render() {
    const CREATION_YEAR = 2017
    const CURR_YEAR     = new Date().getFullYear()
    const navMenuOpen   = this.state.navMenuOpen

    return (
      <aside className="sidebar text-lg-right">
        <div className="sidebar__content container-fluid">
          <Link to="/" className="sidebar__brand">
            <img className="sidebar__brand__image" src={Logo} alt="" width="52" height="52"/>
            {/*<span className="sidebar__brand__name ml-1">Edward Cobbold</span>*/}
            <span className="sidebar__brand__name ml-1">Lorem Ipsum</span>
          </Link>
          {/*<span className="sidebar__brand__job-title hidden-md-down">Frontend Developer</span>*/}
          <span className="sidebar__brand__job-title hidden-md-down">Culpa similique</span>

          <hr className="hidden-md-down"/>

          {/* Mobile nav */}
          <div className="hidden-md-up">
            <button type="button" className="sidebar__btn-sitenav" onClick={this.toggleMenu} aria-expanded={navMenuOpen}>
              <span className="burger-bar"></span>
              <span className="sr-only">Toggle menu</span>
            </button>

            <menu className="sidebar__menu" hidden={!navMenuOpen}>
              <div className="container-fluid">
                <Sitenav/>
              </div>
            </menu>
          </div>

          {/* Tablet+ nav */}
          <div className="hidden-sm-down">
            <Sitenav/>
          </div>
        </div>

        <div className="sidebar__credits hidden-md-down">
          <div className="container-fluid">
            <p>
              <small>&copy; {CURR_YEAR > CREATION_YEAR ?
                <span className="sidebar__credits__date">{CREATION_YEAR}&mdash;{CURR_YEAR}</span> :
                <span className="sidebar__credits__date">{CURR_YEAR}</span>
              }</small>
            </p>
          </div>
        </div>
      </aside>
    )
  } // /render()
} // /export default class Sidebar extends React.Component

export default Sidebar
