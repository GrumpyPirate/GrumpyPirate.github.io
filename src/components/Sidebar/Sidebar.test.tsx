import { shallow } from 'enzyme';
import React from 'react';

import { withMockStore } from 'utils/testing';

import Sidebar from './Sidebar';

describe('Components', () => {
  describe('Sidebar', () => {
    it('should render as expected, without crashing', () => {
      const wrapper = shallow(withMockStore(<Sidebar />));

      expect(wrapper).toMatchSnapshot();
    });
  });
});
