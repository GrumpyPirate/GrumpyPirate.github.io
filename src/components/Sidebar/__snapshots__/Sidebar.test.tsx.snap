// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Sidebar should render as expected, without crashing 1`] = `
.c13 {
  position: absolute;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  padding: 0;
  border: 0;
  height: 1px;
  width: 1px;
  overflow: hidden;
}

.c1 {
  width: 100%;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c16 a {
  font-family: 'Titillium Web',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
  display: inline-block;
  position: relative;
  color: #dc8381;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin-bottom: 0;
  padding-bottom: 0.125em;
}

.c16 a,
.c16 a:hover,
.c16 a:focus,
.c16 a:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c16 a,
.c16 a:hover,
.c16 a:focus,
.c16 a:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c16 a::before {
  background-color: currentColor;
  bottom: 0;
  content: '';
  display: block;
  height: 1px;
  left: 0;
  position: absolute;
  width: 100%;
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition: -webkit-transform 0.15s ease;
  -webkit-transition: transform 0.15s ease;
  transition: transform 0.15s ease;
}

.c16 a:focus::before,
.c16 a:hover::before,
.c16 a:active::before {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.c16 a::before,
.c16 a::after {
  background-color: currentColor;
  bottom: 0;
  content: '';
  display: block;
  height: 1px;
  position: absolute;
  -webkit-transform: scaleX(0);
  -ms-transform: scaleX(0);
  transform: scaleX(0);
  -webkit-transition: -webkit-transform 0.15s ease;
  -webkit-transition: transform 0.15s ease;
  transition: transform 0.15s ease;
  width: 50%;
}

.c16 a::before {
  left: 0;
  -webkit-transform-origin: right center;
  -ms-transform-origin: right center;
  transform-origin: right center;
}

.c16 a::after {
  right: 0;
  -webkit-transform-origin: left center;
  -ms-transform-origin: left center;
  transform-origin: left center;
}

.c16 a:focus::before,
.c16 a:hover::before,
.c16 a:active::before,
.c16 a:focus::after,
.c16 a:hover::after,
.c16 a:active::after {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.c16 a:hover,
.c16 a:focus,
.c16 a.active {
  color: #fff;
}

.c16 a.active::before,
.c16 a.active::after {
  -webkit-transform: scaleX(1);
  -ms-transform: scaleX(1);
  transform: scaleX(1);
}

.c16:not(:last-child) {
  margin: 0 0 0.75rem;
}

.c15 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 3.125rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 0.46875rem;
  padding-top: 0.46875rem;
  position: relative;
  z-index: 11;
}

.c5 {
  background: url(head-sprite.png) top left no-repeat transparent;
  background-size: 2rem 4rem;
  display: block;
  -webkit-filter: drop-shadow(0.25rem 0.25rem 0.125rem rgba(0,0,0,0.2));
  filter: drop-shadow(0.25rem 0.25rem 0.125rem rgba(0,0,0,0.2));
  height: 2rem;
  image-rendering: crisp-edges;
  margin: 0;
  -webkit-transition: -webkit-transform 0.15s;
  -webkit-transition: transform 0.15s;
  transition: transform 0.15s;
  width: 2rem;
}

.c5,
.c5:hover,
.c5:focus,
.c5:active {
  outline: none;
}

.c3 a {
  font-family: 'Titillium Web',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin-bottom: 0;
}

.c3 a,
.c3 a:hover,
.c3 a:focus,
.c3 a:active {
  outline: none;
}

.c3 a,
.c3 a:hover,
.c3 a:focus {
  color: #f2f1f3;
}

.c3 a:hover .c4,
.c3 a:focus .c4 {
  background-position: bottom left;
}

.c3 a:active .c4 {
  -webkit-transform: scale(0.875);
  -ms-transform: scale(0.875);
  transform: scale(0.875);
  -webkit-transition-duration: 0s;
  transition-duration: 0s;
}

.c6 {
  color: inherit;
  display: inline-block;
  margin-left: 1rem;
  vertical-align: middle;
}

.c7 {
  color: #f2f1f3;
  display: block;
  font-size: 0.75em;
  opacity: 0.75;
}

.c8 {
  border-color: rgba(242,241,243,0.2);
  border-style: solid none none;
  border-width: 1px;
  margin: 2rem 0;
}

.c12 {
  top: calc(50% - 0.03125rem);
}

.c12,
.c12::before,
.c12::after {
  background-color: #f2f1f3;
  display: block;
  height: 0.0625rem;
  left: 0;
  position: absolute;
  -webkit-transition: -webkit-transform 0.15s ease,background-color 0.15s ease;
  -webkit-transition: transform 0.15s ease,background-color 0.15s ease;
  transition: transform 0.15s ease,background-color 0.15s ease;
  width: 1.75rem;
}

.c12::before,
.c12::after {
  content: '';
}

.c12::before {
  -webkit-transform: translateY(-0.5rem);
  -ms-transform: translateY(-0.5rem);
  transform: translateY(-0.5rem);
}

.c12::after {
  -webkit-transform: translateY(0.5rem);
  -ms-transform: translateY(0.5rem);
  transform: translateY(0.5rem);
}

.c10 {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  background: transparent;
  border-radius: 0;
  box-shadow: none;
  cursor: pointer;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-appearance: none;
  touch-action: manipulation;
  display: block;
  height: 1.0625rem;
  position: relative;
  -webkit-transition: -webkit-transform 0.075s ease;
  -webkit-transition: transform 0.075s ease;
  transition: transform 0.075s ease;
  width: 1.75rem;
}

.c10::-moz-focus-inner {
  border: 0;
  padding: 0;
}

.c10,
.c10:hover,
.c10:focus,
.c10:active {
  outline: none;
}

.c10:active {
  -webkit-transform: scale(0.9);
  -ms-transform: scale(0.9);
  transform: scale(0.9);
}

.c10[aria-expanded='true'] .c11 {
  background-color: transparent;
}

.c10[aria-expanded='true'] .c11::before {
  -webkit-transform: rotate(-135deg);
  -ms-transform: rotate(-135deg);
  transform: rotate(-135deg);
}

.c10[aria-expanded='true'] .c11::after {
  -webkit-transform: rotate(135deg);
  -ms-transform: rotate(135deg);
  transform: rotate(135deg);
}

.c17 {
  background-color: #67309a;
  border-bottom: solid 0.0625rem #dc8381;
  border-top: solid 0.0625rem rgba(220,131,129,0.25);
  left: 0;
  margin: 0;
  padding: 0.5rem 0;
  position: fixed;
  top: 3.125rem;
  -webkit-transition: opacity 0.15s ease,-webkit-transform 0.15s ease,visibility 0.15s ease;
  -webkit-transition: opacity 0.15s ease,transform 0.15s ease,visibility 0.15s ease;
  transition: opacity 0.15s ease,transform 0.15s ease,visibility 0.15s ease;
  width: 100%;
  z-index: 10;
  opacity: 0;
  pointer-events: none;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  visibility: hidden;
}

.c0 {
  background-color: #67309a;
  color: #f2f1f3;
  left: 0;
  position: fixed;
  right: 0;
  text-transform: uppercase;
  top: 0;
  width: 100%;
  z-index: 11;
}

@media (min-width:48em) {
  .c1 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c16 a {
    font-size: 1rem;
  }
}

@media (min-width:48em) {
  .c16:not(:last-child) {
    margin: 0 0.75rem 0 0;
  }
}

@media (min-width:75em) {
  .c16:not(:last-child) {
    margin: 0;
  }
}

@media (min-width:48em) and (max-width:74.9375em) {
  .c15 {
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:75em) {
  .c2 {
    background: 0;
    display: block;
    height: auto;
    padding-bottom: 0;
    padding-top: 0;
  }
}

@media (min-width:75em) {
  .c5 {
    background-size: 3rem 6rem;
    height: 3rem;
    margin-left: auto;
    width: 3rem;
  }
}

@media (min-width:48em) {
  .c3 a {
    font-size: 1rem;
  }
}

@media (min-width:75em) {
  .c3 a {
    font-size: 1.5rem;
    display: block;
  }
}

@media (max-width:74.9375em) {
  .c7 {
    display: none;
  }
}

@media (max-width:74.9375em) {
  .c8 {
    display: none;
  }
}

@media (min-width:48em) {
  .c9 {
    display: none;
  }
}

@media (max-width:47.9375em) {
  .c14 {
    display: none;
  }
}

@media (min-width:48em) {
  .c17 {
    display: none;
  }
}

@media (min-width:75em) {
  .c0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    border: 0;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    height: 100vh;
    padding: 0;
    position: fixed;
    text-align: right;
    width: 13.75rem;
  }
}

@media (min-width:90em) {
  .c0 {
    width: 16.25rem;
  }
}

<aside
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <a
          href="/"
        >
          <figure
            class="c4 c5"
          />
          <span
            class="c6"
          >
            Edward Cobbold
          </span>
        </a>
      </div>
      <span
        class="c7"
      >
        Frontend Developer
      </span>
      <hr
        class="c8"
      />
      <div
        class="c9"
      >
        <button
          aria-expanded="false"
          class="c10"
          type="button"
        >
          <span
            class="c11 c12"
          />
          <span
            class="c13"
          >
            Toggle menu
          </span>
        </button>
      </div>
      <div
        class="c14"
      >
        <ul
          class="c15"
          role="navigation"
        >
          <li
            class="c16"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              About
            </a>
          </li>
          <li
            class="c16"
          >
            <a
              href="/portfolio"
            >
              Portfolio
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    class="c17"
  >
    <div
      class="c1"
    >
      <ul
        class="c15"
        role="navigation"
      >
        <li
          class="c16"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            About
          </a>
        </li>
        <li
          class="c16"
        >
          <a
            href="/portfolio"
          >
            Portfolio
          </a>
        </li>
      </ul>
    </div>
  </div>
</aside>
`;
